src/Http/Controllers/VoyagerProfileController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace TCG\Voyager\Http\Controllers;
+
+use Illuminate\Http\Request;
+use TCG\Voyager\Events\BreadDataUpdated;
+use TCG\Voyager\Facades\Voyager;
+
+class VoyagerProfileController extends Controller
+{
+    public function index()
+    {
+        return Voyager::view('voyager::profile');
+    }
+
+    // POST BR(E)AD
+    public function update(Request $request)
+    {
+        $slug = 'users';
+
+        $dataType = Voyager::model('DataType')->where('slug', '=', $slug)->first();
+
+        $id = $request->user()->id;
+
+        $data = call_user_func([$dataType->model_name, 'findOrFail'], $id);
+
+        // Check permission
+        $this->authorize('edit', $data);
+
+        //Verify if user tries to modify his own role
+        if ($request->has('role_id')) {
+            $request->merge(['role_id' => $request->user()->role->id]);
+        }
+
+        if($request->has('user_belongstomany_role_relationship')){
+            $request->offsetUnset('user_belongstomany_role_relationship');
+        }
+
```

Comment:
I think you missed the `user_belongsto_role_relationship` field.

```php
if ($request->has('user_belongsto_role_relationship')) {
    $request->offsetUnset('user_belongsto_role_relationship');
}
```

I was able to change my main role to something different by adding `<input type="text" class="form-control" id="user_belongsto_role_relationship" name="user_belongsto_role_relationship" value="1">` to the page.

Reply:
@cyrildewit I think you discovered a new bug.
AFAIK the belogsto (belongstomany is different) is never sent in an edit request because the name is set as:
```php
<select class="form-control select2" name="{{ $options->column }}">
```
in relationship.blade.php, where $option->column is the column referred in the same table (es. role_id)
Therefore the belogsto should never be taken in consideration in the request.